### PR TITLE
Fix #3208

### LINKED ISSUE #3208: Improve write error message
```
r_io_write: cannot write on fd 6
```
* suggest `oo+` or `e io.cache=true`
* show filename
* show address trying to write

### PR DESCRIPTION
**Pull request by alvarofe**

### CODE REVIEW

**Comment by alvarofe** [comment]
Anyone does know why the test is failing? In the laptop I get green in t.tool/rasm2

**Comment by Maijin** [comment]
@alvarofe not related to your fix but on https://github.com/radare/radare2/commit/11acb8392ea0544559b21c89a408a212673633c7

**Comment by condret** [comment]
buäh, every change to r_io will cause lots of works to me

File: libr/io/io.c [hunk 1]
```diff
@@ -647,19 +647,21 @@ R_API int r_io_write(RIO *io, const ut8 *buf, int len) {
...
-		} else { 
+		} else {
 			eprintf ("r_io_write: io handler with no write callback\n");
 			ret = -1;
 		}
 	} else {
-		if (io->desc) {
-			ret = write (io->desc->fd, buf, len);
-		} else ret = -1;
+		if (io->desc) ret = write (io->desc->fd, buf, len);
```
Comment (radare) [inline libr/io/io.c]: Use ternary comparison here

Ret = io->desc? ...:-1;

**Comment by radare** [comment]
Come on. Its just a printf

**Comment by radare** [comment]
Thanks!

**Pull request closed without merging**